seed-job/index_reader.py

```diff
@@ -0,0 +1,173 @@
+"""
+This script parses the container index specified and
+creates the Jenkins pipeline projects from entries of index.
+"""
+
+import os
+import subprocess
+import sys
+import yaml
+
+
+def run_cmd(cmd):
+    """
+    Runs the given command and returns the output and error
+    """
+    return subprocess.Popen(cmd, stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE).communicate()
+
+
+class Project(object):
+    """
+    Class for storing and processing a container index project
+    """
+
+    def __init__(self, entry, namespace):
+        """
+        Initialize project object with an entry in container index
+        """
+        self.load_project_entry(entry)
+        self.namespace = namespace
+
+    def replace_dot_slash_colon_(self, value):
+        """
+        Given a value with either dot slash or underscore,
+        replace each with hyphen
+        """
+        return value.replace("_", "-").replace("/", "-").replace(
+            ".", "-").replace(":", "-")
+
+    def process_depends_on(self, depends_on=None):
+        """
+        Process dependson for given project based on entry index
+        and namespace
+        """
+        if not depends_on:
+            return None
+        return "{}-{}".format(self.namespace,
+                              self.replace_dot_slash_colon_(depends_on)
+                              )
+        """
+        # TODO: process multiple dependson values
+            if dependson is not None:
+                if isinstance(dependson, list):
+                    dependson_job = ','.join(dependson)
+                else:
+                    dependson_job = str(dependson)
+                dependson_job = dependson_job.replace(
+                    ':', '-').replace('/', '-')
+
+            if dependson_job == '':
+                dependson_job = 'none'
+        """
+
+    def process_desired_tag(self, desired_tag=None):
+        """
+        Process desired_tag for given project
+        """
+        if not desired_tag:
+            return "latest"
+        return desired_tag
+
+    def process_build_context(self, build_context=None):
+        """
+        Process build_context for given project
+        """
+        if not build_context:
+            return "./"
+        return build_context
+
+    def process_prebuild_script(self, prebuild_script=None):
+        """
+        Process prebuild_script for given project
+        """
+        if not prebuild_script:
+            return None
+        return prebuild_script
+
+    def process_pre_build_context(self, prebuild_context=None):
+        """
+        Process prebuild_context for given project
+        """
+        if not prebuild_context:
+            return None
+        return prebuild_context
+
+    def load_project_entry(self, entry):
+        """
+        Loads a container index entry in class objects
+        """
+        try:
+            self.appid = self.replace_dot_slash_colon_(entry['app-id'])
```

Comment:
How about pushing all of these constants such as 'app-id' to a constants library, as they might get used in multiple other places

Reply:
as in the key name in each entry ?